server/middleware/auth.js

```diff
@@ -0,0 +1,52 @@
+var router = require('express').Router(),
+    userModel = require('../db/model/User.js');
+
+module.exports = function(checkType) {
+    type = checkType;
+    router.use(function(req, res, next) {
+        if (req.get('Authorization')) {
+            var authorization = req.get('Authorization');
+            var token = authorization.replace(/Bearer /g, '');
+            userModel.find().byToken(token).exec(function(err, user) {
+                if (!err && user) {
+                    var verify = user.verifyToken(token);
+                    if (verify[0]) {
```

Comment:
I could try returning an object. We could also just do it in a procedural callback, which shouldn't be a problem, it's just offloading the code elsewhere